code/server/sv_client.c

```diff
@@ -2156,6 +2156,12 @@ static void SV_UserMove( client_t *cl, msg_t *msg, qboolean delta ) {
 	// in the commands will cause them to be immediately discarded
 	for ( i = 0; i < cmdCount; i++ ) {
 		// if this is a cmd from before a map_restart ignore it
+		if ( cmds[i].serverTime - sv.restartedTime < 0 ) {
+			Com_DPrintf( "ignoring pre-restart usercmd: client(%i) cmdTime(%i) restartedTime(%i)\n",
+					cl - svs.clients, cmds[i].serverTime, sv.restartedTime );
+			continue;
+		}
```

Comment:
This looks redundant, it would be better to re-use existing code path:
```c
if ( cmds[i].serverTime - cl->lastUsercmd.serverTime <= 0 ) {
continue;
}
```
For that purpose `cl->lastUsercmd.serverTime` must be set to last map restart time for <= CS_PRIMED clients if I'm correct

Reply:
This check is intended for CS_ACTIVE clients. It doesn't appear that `cl->lastUsercmd.serverTime` has the correct restart time in this case.

Reply:
I think I know what you're suggesting though, to use the same technique as SV_SendClientGameState and modify lastUsercmd.serverTime during map restart?